app/parse.py

```
@@ -1,12 +1,107 @@
+import logging
+import sys
 from dataclasses import dataclass
+from typing import List
+from urllib.parse import urljoin
+
+import requests
+from bs4 import BeautifulSoup, Tag
+
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s %(levelname)s %(message)s",
+    handlers=[
+        logging.StreamHandler(sys.stdout),
+        logging.FileHandler("parser.log"),
+    ],
+)
+
+BASE_URL = "https://mate.academy/"
 
 
 @dataclass
 class Course:
     name: str
     short_description: str
     duration: str
+    modules: list[str]
+
+
+def get_html(url: str) -> BeautifulSoup:
+    response = requests.get(url)
+    response.raise_for_status()
+    return BeautifulSoup(response.content, "html.parser")
+
+
+def get_course_description(soup: BeautifulSoup) -> str:
+    element = soup.select_one(".SalarySection_aboutProfession__C6ftM")
+    return element.get_text()
+
+
+def get_single_module(module: Tag) -> str:
+    name = module.select_one(".CourseModulesList_topicName__7vxtk").get_text()
+    topics_count = module.select_one(".CourseModulesList_topicsCount__H_fv3")
+    if topics_count:
+        topics_count = topics_count.get_text().split()[0]
+    return f"{name} ({topics_count} topics)"
+
+
+def get_modules(soup: BeautifulSoup) -> list[str]:
+    result = []
+    modules = soup.select(
+        ".CourseModulesList_modulesList__C86yL > .color-dark-blue"
+    )
+    # 1st element is a "get started" text
+    for module in modules[2:]:
+        result.append(get_single_module(module))
```

Avoid the fragile `modules[2:]` slice in `get_modules`. Relying on index offsets silently skips items and is brittle. Either refine the CSS selector to only return real module nodes or filter candidates explicitly (for example check each candidate contains the expected title selector before accepting it). Document the reason if any special filtering is required. This addresses the previous HIGH item about removing/justifying the slice.